modules/gke/main.tf

```diff
@@ -87,6 +91,12 @@ resource "google_container_node_pool" "primary_nodes" {
       "https://www.googleapis.com/auth/cloud-platform"
     ]
 
+    local_ssd_count = var.enable_disk_setup ? var.local_ssd_count : 0
+
+    metadata = {
+      startup_script = var.enable_disk_setup ? local.disk_setup_script : ""
```

Comment:
@jubrad Trying to get lgalloc to work but got a bit stuck on the bootstrap script. I don’t think that you can pass a user data script, startup-script metadata seems to be reserved and Google does not let you override it. As far as I can see, they suggest using a DaemonSet:

https://cloud.google.com/kubernetes-engine/docs/tutorials/automatically-bootstrapping-gke-nodes-with-daemonsets

Reply:
Switching to a daemonset as per the official GCP docs.